src/App.js

```diff
@@ -1,75 +1,131 @@
 import React from 'react';
+import NewTodo from './NewTodo';
+import TodoList from './TodoList';
+import Footer from './Footer';
 
-function App() {
-  return (
-    <section className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
+class App extends React.Component {
+  state = {
+    todos: [],
+    filters: ['all', 'active', 'completed'],
+    valueTitle: '',
+    filterName: '',
+    activeFilterIndex: 0,
+  }
 
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
-        />
-      </header>
+  selectAllHandler = () => {
+    this.setState((state) => {
+      const allSelected = state.todos.every(todo => todo.completed);
 
-      <section className="main" style={{ display: 'block' }}>
-        <input type="checkbox" id="toggle-all" className="toggle-all" />
-        <label htmlFor="toggle-all">Mark all as complete</label>
+      return {
+        todos: state.todos.map(todo => ({
+          ...todo,
+          completed: !allSelected,
+        })),
+      };
+    });
+  }
 
-        <ul className="todo-list">
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-1" />
-              <label htmlFor="todo-1">sdfsdfsdf</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
+  InputChangeHandler = (event) => {
+    this.setState({
+      valueTitle: event.target.value,
+    });
+  };
 
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-2" />
-              <label htmlFor="todo-2">sakgjdfgkhjasgdhjfhs</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
+  addTodoHandler = (event) => {
+    event.preventDefault();
+    if (this.state.valueTitle) {
+      this.setState(state => ({
+        todos: [
+          ...state.todos,
+          {
+            id: +Date.now(),
```

Comment:
Date.now() це число і так
```suggestion
            id: Date.now(),
```